### ISSUE TITLE
Smallfix: up and down arrows in chat box to place cursor in the beginning or end of the message respectively.

### ISSUE THREAD

**Issue by CortexReaver** [OpenRA/OpenRA#20047]
Pressing up and down arrows on keyboard in a chat box will place cursor in the beginning or end of the message respectively.

Just a tiny fix for more pleasant experience, don't think it needs more description. I thought about making a list of small fixes like this to the interface actually, but nothing else is coming to mind. And I don't know if writing several issues in one is considered a good practice here.

**Comment by dragunoff** [comment]
There are already keyboard shortcuts for this: <kbd>Home</kbd> (to start of line) and <kbd>End</kbd> (to end of line). These are de facto standard across operating systems and environments. So I don't think we need another key for that. And in other software Up/Down are usually used for cycling the history of messages (note: we don't keep history at the moment).

**Comment by CortexReaver** [comment]
Yes, but we are talking about chatbox, not a console. If you've been ever typed messages on Youtube, or send a mail via Google mail (or Proton) this is how it works there. This is what people used to.

**Comment by dragunoff** [comment]
> Yes, but we are talking about chatbox, not a console. If you've been ever typed messages on Youtube, or send a mail via Google mail (or Proton) this is how it works there. This is what people used to.

Huh, you're right - it does work that way in most chat applications. It just never occurred to me because I'm so used to Home/End. So, alright then - I think this can be considered.

**Comment by dragunoff** [comment]
Well, on closer inspection - this behavior is a side effect of <kbd>↑</kbd> / <kbd>↓</kbd> moving up and down in a multi-line text field and when there are no more lines to go it goes to the start/end of first/last line. So nothing new here actually. There are no multi-line text inputs in OpenRA so I guess that's why this is not implemented.

**Comment by CortexReaver** [comment]
Hm, and this behavior is slightly different for Youtube (down arrow puts cursor at the END of next line) and mail services (down arrow puts cursor at BEGINNING of the next line). Facebook cursor moving is very similar to notepad on the other hand: moves it like in array or a matrix.  Idk, maybe I am the only weird alien here who uses arrows due text writing. Just got used to them so my "zoomer mentality" repulses navigation keys.
